pkg/controller/clusterprovision/clusterprovision_controller.go

```diff
@@ -456,3 +462,25 @@ func generateOwnershipUniqueKeys(owner hivev1.MetaRuntimeObject) []*controllerut
 		},
 	}
 }
+
+// deleteInstallJob deletes the install job of a successful provision
+func (r *ReconcileClusterProvision) deleteInstallJob(provision *hivev1.ClusterProvision, pLog log.FieldLogger) (reconcile.Result, error) {
+	pLog.Info("deleting successful install job")
+	job := &batchv1.Job{}
+	if err := r.Get(context.TODO(), types.NamespacedName{Name: provision.Status.JobRef.Name, Namespace: provision.Namespace}, job); err != nil {
+		pLog.WithError(err).Log(controllerutils.LogLevel(err), "could not get install job")
+	}
+	// deleting install job with background propagation policy to cascade delete install pod
+	if err := r.Delete(context.TODO(), job, client.PropagationPolicy(metav1.DeletePropagationBackground)); err != nil {
+		pLog.WithField("job", job.Name).WithError(err).Log(controllerutils.LogLevel(err), "error deleting successful install job")
+		return reconcile.Result{}, err
+	}
```

Comment:
`Delete` can only be called if there was not an error returned from `Get`. Otherwise, the `Delete` call will error.

```suggestion
	switch err := r.Get(context.TODO(), types.NamespacedName{Name: provision.Status.JobRef.Name, Namespace: provision.Namespace}, job); {
	case apierrors.IsNotFound(err):
		pLog.Info("install job has already been deleted")
	case err != nil:
		pLog.WithError(err).Log(controllerutils.LogLevel(err), "could not get install job")
		return reconcile.Result{}, err
	default:
		// deleting install job with background propagation policy to cascade delete install pod
		if err := r.Delete(context.TODO(), job, client.PropagationPolicy(metav1.DeletePropagationBackground)); err != nil {
			pLog.WithField("job", job.Name).WithError(err).Log(controllerutils.LogLevel(err), "error deleting successful install job")
			return reconcile.Result{}, err
		}
	}
```